main.tf

```diff
@@ -45,6 +54,13 @@ resource "azurerm_storage_account" "storage_account" {
   )
 }
 
+resource "azurerm_role_assignment" "storage-account-role-assignment" {
+  count                = length(local.role_assignments)
```

Comment:
better to use `for_each` here so the role assignment is keyed by the name of the role in the state and not by its position in the list.

Otherwise someone re-ordering the list or adding a new role to the beginning of the list will trigger the role to be deleted and recreated. Not a huge problem in this scenario but it could cause the MI to lose access for a period of time and this sort of delete and re-create is vulnerable to errors on the azure side rejecting it and then causing the whole run to fail...

you can see the 'key' in the terraform plan currently is:
> module.storage_account.azurerm_role_assignment.storage-account-send-letter-blob-delegator-role[1]